pkg/ns/ns_unspecified.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2015-2017 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !linux
+
+package ns
+
+func getCurrentThreadNetNSPath() string {
```

Comment:
This makes me a bit uncomfortable: this is used in `ns.Do()`. I know that it currently won't cause problems, but...
Could you refactor `Do()` to use `GetCurrentNS()`, move `GetCurrentNS()` to the per-os files, and move `getCurrentThreadNetNSPath` to `ns_linux.go`?